src/Microsoft.ML.Core/Utilities/Utils.cs

```diff
@@ -966,6 +966,14 @@ private static MethodInfo MarshalInvokeCheckAndCreate<TRet>(Type genArg, Delegat
             return meth;
         }
 
+        private static MethodInfo MarshalInvokeCheckAndCreate<TRet>(Delegate func, Type[] genArgs)
+        {
+            var meth = MarshalActionInvokeCheckAndCreate(func, genArgs);
+            if (meth.ReturnType != typeof(TRet))
+                throw Contracts.ExceptParam(nameof(func), "Cannot be generic on return type");
+            return meth;
```

Comment:
I don't think `method` is a reserved keyword in C#. Recommend the longer name.
```suggestion
            return method;
```

Reply:
I kept it as `meth` because that is consistent with all the other methods here that do this same thing. I think if we want it to be `method` then we need to change them all.

---
In reply to: [347619466](https://github.com/dotnet/machinelearning/pull/4413#discussion_r347619466) [](ancestors = 347619466)